bookmaker_titlepage.rb

```diff
@@ -93,42 +91,223 @@ def getResourceDir(imprint, json)
+## wrapping a Mcmlln::Tools method in a new method for this script; to return a result for json_logfile
+def makeFolder(path, logkey='')
+  unless Dir.exist?(path)
+    Mcmlln::Tools.makeDir(path)
+  else
+    logstring = 'n-a'
+  end
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-puts "RUNNING TITLEPAGEMAKER"
+def checkForOldTitlepages(coverdir, logkey='')
+  oldtitlepage = ""
+  oldepubtitlepage = File.join(coverdir, "epubtitlepage.jpg")
+  oldpodtitlepage = File.join(coverdir, "titlepage.jpg")
+
+  if File.file?(oldepubtitlepage)
+    logstring = "Found an archived EPUB titlepage"
+    oldtitlepage = oldepubtitlepage
+  elsif File.file?(oldpodtitlepage)
+    logstring = "Found an archived POD titlepage"
+    oldtitlepage = oldpodtitlepage
+  end
+  puts logstring
+  return oldtitlepage
+rescue => logstring
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-# paths to key scripts and JSON metadata
-pdftmp_dir = File.join(Bkmkr::Paths.project_tmp_dir_img, "pdftmp")
-pdfmaker_dir = File.join(Bkmkr::Paths.core_dir, "bookmaker_pdfmaker")
-imprint_json = File.join(Bkmkr::Paths.scripts_dir, "bookmaker_addons", "imprints.json")
+def checkForSubmittedTitlepages(submitted_images, logkey='')
+  allimg = File.join(submitted_images, "*")
+  etparr = Dir[allimg].select { |f| f.include?('epubtitlepage.')}
+  ptparr = Dir[allimg].select { |f| f.include?('titlepage.')}
 
-configfile = File.join(Bkmkr::Paths.project_tmp_dir, "config.json")
-data_hash = Mcmlln::Tools.readjson(configfile)
+  newtitlepage = ""
 
-# the cover filename and metadata
-project_dir = data_hash['project']
-stage_dir = data_hash['stage']
-resource_dir = data_hash['resourcedir']
+  if etparr.any?
+    logstring = "Found a new EPUB titlepage"
+    newtitlepage = etparr.find { |e| /[\/|\\]epubtitlepage\./ =~ e }
+  elsif ptparr.any?
+    logstring = "Found a new POD titlepage"
+    newtitlepage = ptparr.find { |e| /[\/|\\]titlepage\./ =~ e }
+  end
+  puts logstring
+  return newtitlepage
+rescue => logstring
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-# --------------- ISBN FINDER COPIED FROM BOOKMAKER_ADDONS/METADATA_PREPROCESSING
-# testing to see if ISBN style exists
-spanisbn = File.read(Bkmkr::Paths.outputtmp_html).scan(/spanISBNisbn/)
-multiple_isbns = File.read(Bkmkr::Paths.outputtmp_html).scan(/spanISBNisbn">\s*.+<\/span>\s*\(((hardcover)|(trade\s*paperback)|(mass.market.paperback)|(print.on.demand)|(e\s*-*\s*book))\)/)
+def setFinalCover(oldtitlepage, newtitlepage, logkey='')
+  final_cover = oldtitlepage
+  if File.file?(newtitlepage)
+    final_cover = newtitlepage
+  end
+  return final_cover
+rescue => logstring
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+# Determine whether or not to generate a titlepage.
+def decideTitlepageGeneration(titlepagelog, newtitlepage, oldtitlepage, final_cover, logkey='')
```

Comment:
rename to "detectAutoGeneratedTitlepage" to match covermaker function name (or vice versa)